File: lcci/0108-zero-matrix/solution.rs
impl Solution {
    pub fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
        let nrow = matrix.len();
        let ncol = matrix[0].len();
        let mut rows: Vec<bool> = vec![false; nrow];
        let mut cols: Vec<bool> = vec![false; ncol];
        
        for i in 0..nrow {
            for j in 0..ncol {
                if matrix[i][j] == 0 {
                    rows[i] = true;
                    cols[j] = true;
                }
            }
        }
        
        for i in 0..nrow {
            for j in 0..ncol {
                if rows[i] || cols[j] {
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: lcci/0108-zero-matrix/solution.cc
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int n = matrix.size(), m = matrix[0].size();
        vector<bool> row(n), col(m);
        for (int r = 0; r < n; ++r) {
            for (int c = 0; c < m; ++c) {
                if (matrix[r][c] == 0) {
                    row[r] = true;
                    col[c] = true;
                }
            }
        }

        for (int r = 0; r < n; ++r) {
            for (int c = 0; c < m; ++c) {
                if (row[r] || col[c]) {
                    matrix[r][c] = 0;
                }
            }
        }
    }
};

